src/App.tsx

```diff
@@ -1,27 +1,54 @@
-import React from 'react';
+import React, { useReducer } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+type Action = {
+  payload: Good[]
+};
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const reducer = (state: Good[], action: Action) => {
+  if (action.payload) {
```

Comment:
In this case it doesn't matter that much, but if would you have more actions it's better to use switch. 
A few notes: 
- in typical reducer usage both in react alone and react with redux state, you want to handle different operation on your state like resetting, deleting, adding etc. based on type of action
- if you had a couple of ifs, js interpreter would go from top to bottom through your code until it would find a satisfying result, in switch statement it immediately jumps to right case. For example we have a structure like below and want to dispatch an action of type 'setBooleansOpposite'


![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/87986955/4950ee20-7563-46c0-a8fa-5359a615b719)

### in this reducer we will first check every if before reaching 'setBooleansOpposite'
![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/87986955/e1fc100a-397d-42d2-9d37-617153c6eef9)

### in this reducer we will immediately jump into 'setBooleanOpposite'
![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/87986955/a2bf5794-7ed5-41d7-8421-36fa6104ac34)

Reply:
Generally reducers are more suitable for handling complex states, in this task a simple useState would suffice,  however remember for future reference similar pattern if you have to handle something more complex